Zend/zend_vm_def.h

```diff
@@ -4248,13 +4248,17 @@ ZEND_VM_INLINE_HANDLER(62, ZEND_RETURN, CONST|TMP|VAR|CV, ANY, SPEC(OBSERVER))
 		if (return_value) {
 			ZVAL_NULL(return_value);
 		}
+		ZEND_OBSERVER_SAVE_OPLINE();
+		ZEND_OBSERVER_FCALL_END(execute_data, retval_ptr);
```

Comment:
Might it make sense to just move this to the very start of the function, rather than duplication it in the branches?

Reply:
That would make sense and would be simpler. But the main issue is that any errors that are raised in the return handler would trigger after the end handler had fired.

For example, the output from `observer_retval_05.phpt` would change to this:

```diff
  <foo>
+  </foo:NULL>

Warning: Undefined variable $i_do_not_exist in %s on line %d
-  </foo:NULL>
```

For the PHP tracer extension I'm working on, that would mean that we couldn't attach the `E_WARNING` to the open `foo` span that raised the error.